internal/cli/workspace_create.go

```diff
@@ -0,0 +1,95 @@
+package cli
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	"github.com/hashicorp/waypoint/internal/clierrors"
+	"github.com/hashicorp/waypoint/internal/pkg/flag"
+	pb "github.com/hashicorp/waypoint/internal/server/gen"
+	"github.com/posener/complete"
+)
+
+type WorkspaceCreateCommand struct {
+	*baseCommand
+
+	flagWorkspaceName string
+}
+
+func (c *WorkspaceCreateCommand) Run(args []string) int {
+	// Initialize. If we fail, we just exit since Init handles the UI.
+	flagSet := c.Flags()
+	if err := c.Init(
+		WithArgs(args),
+		WithFlags(flagSet),
+		WithNoConfig(),
+	); err != nil {
+		return 1
+	}
+	args = flagSet.Args()
+
+	if len(args) != 1 {
+		c.ui.Output(c.Help(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	workspaceName := args[0]
+
+	client := c.project.Client()
+	resp, err := client.UpsertWorkspace(c.Ctx, &pb.UpsertWorkspaceRequest{
+		Workspace: &pb.Workspace{
+			Name: workspaceName,
+		},
+	})
+	if err != nil {
+		c.ui.Output(
+			clierrors.Humanize(err),
+			terminal.WithErrorStyle(),
+		)
+		return 1
+	}
+
+	// this is unlikely to happen with a nil error above, but added here to be
+	// defensive.
+	if resp.Workspace == nil {
+		c.ui.Output(
+			fmt.Sprintf("no workspace returned for create command with name %q", workspaceName),
```

Comment:
The `Output` function can format so so need for `fmt.Sprintf` here

Reply:
Updated in c169c87b5